src/engines/mod.rs

```diff
@@ -383,7 +377,9 @@ impl<'a, I: 'a + IoProvider> ExecutionState<'a, I> {
             }
         }
 
-        panic!("unexpected handle {:?}", handle);
+        tt_warning!(self.status, "input close failed: unexpected handle {:?}", handle);
```

Comment:
Do you want me to add "serious internal bug" only to this particular error, or to others as well?